services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/OAuth2ResourceServerSecurityConfiguration.kt

```diff
@@ -24,11 +36,68 @@ class OAuth2ResourceServerSecurityConfiguration(
     @Bean
     fun filterChain(http: HttpSecurity): SecurityFilterChain {
         AccessRules.configureHttpAccessRules(http)
-        http.oauth2ResourceServer { resourceServer: OAuth2ResourceServerConfigurer<HttpSecurity?> ->
+        http.oauth2ResourceServer { resourceServer ->
             resourceServer.opaqueToken { opaqueToken -> opaqueToken.introspector(introspector()) }
         }
         return http.build()
     }
+
+    /**
+     * Higher priority filter chain for authenticating GDPR API requests.
+     *
+     * When GDPR API is enabled, i.e. haitaton.gdpr.disabled is false, all requests to the GDPR API
+     * endpoints are authenticated by parsing and validating the JWT in the request.
+     */
+    @Bean
+    @Order(2)
```

Comment:
How does this work, should there be a bean with order 1 somewhere?

Reply:
The beans with higher priorities (lower number) are considered first. If there's no order, the bean is regarded as the lowest priority and gets an Order of `Integer.MAX_VALUE`.

While developing, I had a test bean with `@Order(1)`, but I didn't need it after all.

It doesn't matter what the value is, as long as it's smaller than `Integer.MAX_VALUE`, but it might be better to change this to `@Order(1)`, so a reader doesn't look for it elsewhere.